examples/llm_distill/main.py

```diff
@@ -117,7 +107,9 @@ def train():
 
     if model_args.single_model:
         logger.info("Loading single model only...")
-        model = _teacher_factory(model_path)
+        model = transformers.AutoModelForCausalLM.from_pretrained(
+            model_path, device_map=PartialState().process_index
```

Comment:
why is `device_map` needed here? Without the device map, the model will be loaded in CPU (indistributed mode, all ranks except rank 0 will be loaded in meta device).
 ```suggestion
            model_path, torch_dtype=torch.bfloat16()
```